src/main/java/eu/dissco/core/digitalspecimenprocessor/service/MidsService.java

```diff
@@ -54,88 +57,175 @@ private boolean compliesToMidsTwo(DigitalSpecimen attributes) {
     }
   }
 
+  private boolean compliesToMidsThree(DigitalSpecimen attributes) {
+    var topComplies = hasValue(attributes.getDwcInstitutionId());
+    var taxValuesComply = taxValuesComply(attributes);
+    var geographicalValuesComply = geographicalValuesComply(attributes);
+    if (PALEO_GEO_DISCIPLINES.contains(attributes.getOdsTopicDiscipline())) {
+      return topComplies && taxValuesComply && geographicalValuesComply;
+    } else {
+      return topComplies && taxValuesComply && geographicalValuesComply && isValid(
+          attributes.getDwcRecordedById());
+    }
+  }
+
+  private boolean geographicalValuesComply(DigitalSpecimen attributes) {
+    if (attributes.getDwcOccurrence() != null
+        && attributes.getDwcOccurrence().get(0).getDctermsLocation() != null) {
+      var location = attributes.getDwcOccurrence().get(0).getDctermsLocation();
+      if (isValid(location.getDwcLocationID())) {
+        return true;
+      } else if (location.getGeoReference() != null) {
+        var georeference = location.getGeoReference();
+        return (georeference.getDwcDecimalLatitude() != null
+            && georeference.getDwcDecimalLongitude() != null
+            && isValid(georeference.getDwcGeodeticDatum())
+            && georeference.getDwcCoordinateUncertaintyInMeters() != null
+            && georeference.getDwcCoordinatePrecision() != null)
+            || (isValid(georeference.getDwcFootprintWkt())
+            && isValid(georeference.getDwcFootprintSrs()));
+      }
+    }
+    return false;
+  }
+
+  private boolean taxValuesComply(DigitalSpecimen attributes) {
+    var hasScientificNameId = false;
+    var hasIdentifiedById = false;
+    for (var identifications : attributes.getDwcIdentification()) {
+      if (isValid(identifications.getDwcIdentificationID())) {
+        hasIdentifiedById = true;
+      }
+      for (var taxonIdentification : identifications.getTaxonIdentifications()) {
+        if (isValid(taxonIdentification.getDwcScientificNameId())) {
+          hasScientificNameId = true;
+        }
+      }
+    }
+    return hasScientificNameId && hasIdentifiedById;
+  }
+
   private boolean compliesToMidsTwoBio(DigitalSpecimen attributes) {
-    return (attributes.getOdsMarkedAsType() != null && attributes.getOdsMarkedAsType())
+    return attributes.getOdsMarkedAsType() != null
         && (attributes.getOdsHasMedia() != null && attributes.getOdsHasMedia())
```

Comment:
how about `Boolean.TRUE.equals(attributes.getOdsHasMedia())`, then you don't need to do the null check

Reply:
:+1: